test/refunder-test.ts

```diff
@@ -0,0 +1,295 @@
+		it("Not owner should send ethers to Refunder", async function() {
+			const balanceOfRefunderBefore = await ethers.provider.getBalance(refunder.address);
+			const value = ethToWei('0.5');
+			const res = await addr1.sendTransaction({
+				value: value,
+				to: refunder.address
+			});
+	
+			await res.wait();
+	
+			const balanceOfRefunderAfter = await ethers.provider.getBalance(refunder.address);
+	
+			expect(balanceOfRefunderBefore.toString()).to.be.not.eq(balanceOfRefunderAfter.toString());
+			expect(value.toString()).to.be.eq(balanceOfRefunderAfter.toString());
+		});
+	});
+
+	describe('Withdraw ETHs', () => {
+		it("Owner should withdraw ETHs from Refunder", async function() {
+		
+			const value = ethToWei('2');
+			const res = await owner.sendTransaction({
+				value: value,
+				to: refunder.address
+			});
+	
+			await res.wait();
+
+			const balanceOfRefunderBefore = await ethers.provider.getBalance(refunder.address);
+
+			const withdrawRes = await refunder.withdraw(value);
+			await withdrawRes.wait();
+
+			const balanceOfRefunderAfter = await ethers.provider.getBalance(refunder.address);
+	
+			expect(value.toString()).to.be.eq(balanceOfRefunderBefore.toString());
+			expect('0').to.be.eq(balanceOfRefunderAfter.toString());
+		});
+
+		it("Owner should NOT withdraw ETHs from Refunder if balance is 0", async function() {
+		
+			const value = ethToWei('2');
+			const balanceOfRefunderBefore = await ethers.provider.getBalance(refunder.address);
+
+			expect('0').to.be.eq(balanceOfRefunderBefore.toString());
+			await expect(refunder.withdraw(value)).to.be.revertedWith(INSUFFICIENT_BALANCE);
+
+		});
+	
+		it("Not Owner should NOT withdraw ETHs from Refunder", async function() {
+			const value = ethToWei('2');
+			const res = await owner.sendTransaction({
+				value: value,
+				to: refunder.address
+			});
+	
+			await res.wait();
+
+			await expect(refunder.connect(addr1).withdraw(value)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+
+		it("Not Owner should NOT withdraw ETHs from Refunder if balance is 0", async function() {
+		
+			const value = ethToWei('2');
+			const balanceOfRefunderBefore = await ethers.provider.getBalance(refunder.address);
+
+			expect('0').to.be.eq(balanceOfRefunderBefore.toString());
+			await expect(refunder.connect(addr1).withdraw(value)).to.be.revertedWith(NOT_AN_OWNER);
+
+		});
+	});
+
+	describe('Max Gas Price', () => {
+		it("Owner should get max gas price", async function() {
+			const gasPrice = await refunder.maxGasPrice();
+			
+			expect('0').to.be.eq(gasPrice.toString());
+		});
+
+		it("Owner should set max gas price", async function() {
+			const gasPrice = await refunder.maxGasPrice();
+
+			const value = 152;
+
+			let res = await refunder.setMaxGasPrice(value);
+			await res.wait();
+
+			const updatedGasPrice = await refunder.maxGasPrice();
+
+			expect(updatedGasPrice.toString()).to.be.eq(value.toString());
+
+		});
+	
+		it("Not Owner should get max gas price", async function() {
```

Comment:
I think that the `getter` for maxGasPrice is tested in the above test. The test is unnecessary imo